src/app/App.tsx

```diff
@@ -443,6 +471,42 @@ class App extends AsyncComponent<RouteComponentProps, AppState> {
 
         document.title = title;
     }
+
+    /**
+     * Refresh the token one minute before it expires.
+     */
+    private validateTokenPeriodically() {
+        if (this._tokenExpiryTimer !== undefined) {
+            clearInterval(this._tokenExpiryTimer);
+            this._tokenExpiryTimer = undefined;
```

Comment:
Here when you write three lines, and you notice that you wrote those exact same three lines already above (L256), BOOM 💥 ...a new function is born 👶🏻 .